### PR TITLE
mon: Switch to new style options

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

File: src/common/legacy_config_opts.h [hunk 1]
```diff
@@ -188,128 +178,7 @@ OPTION(ms_dpdk_debug_allow_loopback, OPT_BOOL)
 OPTION(ms_dpdk_rx_buffer_count_per_core, OPT_INT)
 
 OPTION(inject_early_sigterm, OPT_BOOL)
-
-OPTION(mon_data, OPT_STR)
-OPTION(mon_initial_members, OPT_STR)    // list of initial cluster mon ids; if specified, need majority to form initial quorum and create new cluster
```
Comment (tchaikov) [inline src/common/legacy_config_opts.h]: please move the descriptions to `options.cc`.

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
@tchaikov Please check, when you get a chance.

File: src/mon/Monitor.cc [hunk 2]
```diff
@@ -1184,7 +1184,7 @@ void Monitor::sync_start(entity_inst_t &other, bool full)
 
     store->apply_transaction(t);
 
-    assert(g_conf->mon_sync_requester_kill_at != 1);
+    assert(g_conf->get_val<int64_t>("mon_sync_requester_kill_at") != 1);
```
Comment (tchaikov) [inline src/mon/Monitor.cc]: @shinobu-x could you remove the migrated config options from `legacy_config_opts.h`?

**Comment by shinobu-x** [comment]
test this please

File: src/common/legacy_config_opts.h [hunk 3]
```diff
@@ -188,128 +178,7 @@ OPTION(ms_dpdk_debug_allow_loopback, OPT_BOOL)
...
-OPTION(mon_reweight_min_bytes_per_osd, OPT_U64)   // min bytes per osd for reweight-by-utilization command
-OPTION(mon_reweight_max_osds, OPT_INT)   // max osds to change per reweight-by-* command
-OPTION(mon_reweight_max_change, OPT_DOUBLE)
-OPTION(mon_health_to_clog, OPT_BOOL)
-OPTION(mon_health_to_clog_interval, OPT_INT)
-OPTION(mon_health_to_clog_tick_interval, OPT_DOUBLE)
-OPTION(mon_health_preluminous_compat, OPT_BOOL)
-OPTION(mon_data_avail_crit, OPT_INT)
-OPTION(mon_data_avail_warn, OPT_INT)
-OPTION(mon_data_size_warn, OPT_U64) // issue a warning when the monitor's data store goes over 15GB (in bytes)
```
Comment (tchaikov) [inline src/common/legacy_config_opts.h]: again, the option descriptions are lost while these options are removed from `legacy_config_opts.h`.

Comment (shinobu-x) [reply src/common/legacy_config_opts.h]: @tchaikov Please check, when you get a chance.

File: src/common/legacy_config_opts.h [hunk 4]
```diff
@@ -188,128 +178,7 @@ OPTION(ms_dpdk_debug_allow_loopback, OPT_BOOL)
...
-OPTION(mon_tick_interval, OPT_INT)
-OPTION(mon_session_timeout, OPT_INT)    // must send keepalive or subscribe
-OPTION(mon_subscribe_interval, OPT_DOUBLE)  // for legacy clients only
-OPTION(mon_delta_reset_interval, OPT_DOUBLE)   // seconds of inactivity before we reset the pg delta to 0
-OPTION(mon_osd_laggy_halflife, OPT_INT)        // (seconds) how quickly our laggy estimations decay
-OPTION(mon_osd_laggy_weight, OPT_DOUBLE)          // weight for new 'samples's in laggy estimations
-OPTION(mon_osd_laggy_max_interval, OPT_INT)      // maximum value of laggy_interval in laggy estimations
-OPTION(mon_osd_adjust_heartbeat_grace, OPT_BOOL)    // true if we should scale based on laggy estimations
-OPTION(mon_osd_adjust_down_out_interval, OPT_BOOL)  // true if we should scale based on laggy estimations
-OPTION(mon_osd_auto_mark_in, OPT_BOOL)         // mark any booting osds 'in'
```
Comment (tchaikov) [inline src/common/legacy_config_opts.h]: @shinobu-x not all descriptions migrated.

**Comment by shinobu-x** [comment]
@tchaikov Please check when you get a chance.

File: src/common/legacy_config_opts.h (hunk 4 above)
Comment (tchaikov) [reply src/common/legacy_config_opts.h]: @shinobu-x this comment is not addressed. please ping me again, once you've finished doing so.

**Comment by jdurgin** [comment]
This would need a rebase now. Moving to non-legacy options means they're also no longer injectable, so I'm closing this. Please re-open if desired.

**Pull request closed without merging**